netty/src/main/java/io/grpc/transport/netty/GrpcSslContexts.java

```diff
@@ -47,15 +50,30 @@
 public class GrpcSslContexts {
   private GrpcSslContexts() {}
 
-  private static ApplicationProtocolConfig DEFAULT_APN = new ApplicationProtocolConfig(
-          Protocol.ALPN,
-          SelectorFailureBehavior.FATAL_ALERT,
-          SelectedListenerFailureBehavior.FATAL_ALERT,
-          "h2",
-          "h2-17",
-          "h2-16",
-          "h2-15",
-          "h2-14");
+  private static String[] HTTP2_VERSIONS = {
+      "h2",
+      "h2-17",
+      "h2-16",
+      "h2-15",
+      "h2-14"};
+
+  public static ApplicationProtocolConfig ALPN = new ApplicationProtocolConfig(
```

Comment:
> To require every client to have to define their own ApplicationProtocolConfig for this in order to use OpenSSL seems like a pain point.

I don't really follow. The QPS benchmark is making its own `ApplicationProtocolConfig` in this change using this config... So... I don't understand you can argue that exposing this config prevents applications from needing to create their own `ApplicationProtocolConfig`.

I'd assume we just have an overloaded configure() method that accepts the SslProvider to be used (and not expose `selectApplicationProtocolConfig()`); it would have enough information to choose the appropriate ApplicationProtocolConfig. But even with the current code, `selectApplicationProtocolConfig()` returns the appropriate `ApplicationProtocolConfig`, so why isn't that enough? Why do these configs need to be public?